### PR TITLE
Refacto UI : modify http error management in server classes

### PR DESCRIPTION
**Pull request by freddidierRTE**
Add a ServerResponse class to manage error in business services without having to be coupled with angular HttpClient 

Nothing in release note

### CODE REVIEW

File: ui/main/src/app/business/services/processes.service.spec.ts [hunk 1]
```diff
@@ -81,13 +80,22 @@ describe('Processes Services', () => {
         });
     });
 
-    describe('#queryProcess', () => {
-        const processDefinition = new Process('testPublisher', '0', 'businessconfig.label');
-        it('should load businessconfig from remote server', () => {
-            processServerMock.setProcessDefinition(processDefinition);
+    describe('Query a process', () => {
+        it('GIVEN a process exist WHEN query the process THEN process is return', () => {
```
Comment (vlo-rte) [inline ui/main/src/app/business/services/processes.service.spec.ts:84]: What do you think of 'GIVEN an existing process WHEN query the process THEN process is returned' ?

File: ui/main/src/app/business/services/processes.service.spec.ts [hunk 2]
```diff
@@ -81,13 +80,22 @@ describe('Processes Services', () => {
...
+            const processDefinition = new Process('testPublisher', '0', 'businessconfig.label');
+            const serverResponse = new ServerResponse<Process>(processDefinition, null, null);
+            processServerMock.setResponseForProcessDefinition(serverResponse);
             processesService
                 .queryProcess('testPublisher', '0')
                 .subscribe((result) => expect(result).toEqual(processDefinition));
         });
+
+        it('GIVEN a process does not exist WHEN query the process THEN null return', () => {
```
Comment (vlo-rte) [inline ui/main/src/app/business/services/processes.service.spec.ts:93]: What do you think of 'GIVEN a non-existing process WHEN query the process THEN null is returned' ?

**Review by vlo-rte — CHANGES_REQUESTED** [review]

File: ui/main/src/app/server/angular.server..ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    protected processHttpResponse<dataType>(httpResponse: Observable<dataType>): Observable<ServerResponse<dataType>> {
+        return httpResponse.pipe(
+            map((data) => {
+                return new ServerResponse(data, ServerResponseStatus.OK, '');
+            }),
+            catchError((error) => {
+                let serverStatus = ServerResponseStatus.UNKNOWN_ERROR;
+                if (error.status === 404) serverStatus = ServerResponseStatus.NOT_FOUND;
+                else if (error.status === 403) serverStatus = ServerResponseStatus.FORBIDDEN;
+
```
Comment (quinarygio) [inline ui/main/src/app/server/angular.server..ts:23]: I wonder if it could be useful to log the error status when not  404 or 403

Comment (freddidierRTE) [reply ui/main/src/app/server/angular.server..ts:23]: the log is already present in the network panel so i do not think we have to repeat it here

**Review by vlo-rte — APPROVED** [review]

**Pull request merged**